File: HealthCare.Admin/Services/AuthService.cs
using System.Net.Http.Json;
using HealthCare.Shared;
using HealthCare.Shared.DTOs.Auth;

namespace HealthCare.Admin.Services
{
    public class AuthService
    {
        private readonly HttpClient _http;
        private readonly TokenService _tokenService; 

        public AuthService(HttpClient http, TokenService tokenService)
        {
            _http = http;
            _tokenService = tokenService;
        }

        public async Task LoginAsync(LoginDto loginDto)
        {
            var response = await _http.PostAsJsonAsync("api/auth/login", loginDto);
            response.EnsureSuccessStatusCode();

            var result = await response.Content.ReadFromJsonAsync<ApiResponse<AuthResponseDto>>();
            if (result?.Data != null)
            {
                await _tokenService.SetToken(result.Data.AccessToken);
            }
        }
    }
}

File: HealthCare.Portal/src/app/core/services/auth.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, tap, map } from 'rxjs';
import { AuthResponse, LoginDto, RegisterPatientDto } from '../models/auth.models';
import { ApiResponse } from '../models/api-response';
import { TokenService } from './token.service';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class AuthService {
  private readonly baseUrl = `${environment.apiUrl}/auth`;

  constructor(
    private readonly http: HttpClient,
    private readonly tokenService: TokenService,
  ) {}

  login(dto: LoginDto): Observable<AuthResponse> {
    return this.http.post<ApiResponse<AuthResponse>>(`${this.baseUrl}/login`, dto)
      .pipe(
        tap(wrapper => {
          if (wrapper.data?.accessToken) {
            this.tokenService.saveToken(wrapper.data.accessToken);
          }
        }),
        map(wrapper => wrapper.data)
      );
  }

  registerPatient(dto: RegisterPatientDto): Observable<Object> {
    return this.http.post(`${this.baseUrl}/register/patient`, dto);
  }

  logout(): void {
    this.tokenService.removeToken();
  }

  isLoggedIn(): boolean {
    return this.tokenService.isLoggedIn();
  }

  getRole(): string | null {
    return this.tokenService.getRole();
  }
}
